trace/http-out-test-cases.json

```diff
@@ -0,0 +1,274 @@
+[
+    {
+      "name": "Successful GET call to https://example.com",
+      "method": "GET",
+      "url": "https://example.com/",
+      "spanName": "/",
+      "spanStatus": "OK",
+      "spanKind": "Client",
+      "spanAttributes": {
+        "http.path": "/",
+        "http.method": "GET",
+        "http.host": "example.com",
+        "http.status_code": "200",
+        "http.url": "https://example.com/"
+      }
+    },
+    {
+      "name": "Successfully POST call to https://example.com",
+      "method": "POST",
+      "url": "https://example.com/",
+      "spanName": "/",
+      "spanStatus": "OK",
+      "spanKind": "Client",
+      "spanAttributes": {
+        "http.path": "/",
+        "http.method": "POST",
+        "http.host": "example.com",
+        "http.status_code": "200",
+        "http.url": "https://example.com/"
+      }
+    },
+    {
+      "name": "Name is populated as a path",
+      "method": "GET",
+      "url": "http://{host}:{port}/path/to/resource/",
+      "responseCode": 200,
+      "spanName": "/path/to/resource/",
+      "spanStatus": "OK",
+      "spanKind": "Client",
+      "spanAttributes": {
+        "http.path": "/path/to/resource/",
+        "http.method": "GET",
+        "http.host": "{host}:{port}",
+        "http.status_code": "200",
+        "http.url": "http://{host}:{port}/path/to/resource/"
+      }
+    },
+    {
+      "name": "Call that cannot resolve DNS will be reported as error span",
+      "method": "GET",
+      "url": "https://sdlfaldfjalkdfjlkajdflkajlsdjf.sdlkjafsdjfalfadslkf.com/",
+      "spanName": "/",
+      "spanStatus": "UNKNOWN",
+      "spanKind": "Client",
+      "spanAttributes": {
+        "http.path": "/",
+        "http.method": "GET",
+        "http.host": "sdlfaldfjalkdfjlkajdflkajlsdjf.sdlkjafsdjfalfadslkf.com",
+        "http.url": "https://sdlfaldfjalkdfjlkajdflkajlsdjf.sdlkjafsdjfalfadslkf.com/"
+      }
+    },
+    {
+      "name": "Response code: 199. This test case is not possible to implement for some platforms as this considered an invalid status code. Fallback into 200 test case",
```

Comment:
not sure what this means. spanStatus should be "OK' if it cannot be implemented and "UNKNOWN" if it can be implemented?

Reply:
on some platforms this test case is impossible to implement. I kept it as an entry, but actual parameters are for `200` test case